src/typing_extensions.py

```diff
@@ -1,3 +1,4 @@
+# pyright: ignore[reportShadowedImports]
```

Comment:
we don't run pyright in CI, so I'd prefer not to have pyright pragma comments in our code. (I know it breaks pyright's brain a little bit if you open `typing_extensions` or CPython in VSCode; I tend to just use a text editor or IDLE when working on these projects).

Reply:
Yes sorry this is an oversight, I used it in development to avoid having pyright emitting warnings on all the file:
![image](https://github.com/user-attachments/assets/03768a05-16e2-4baf-a223-41478f275507)